commit af004276fd8f6c305727183c159b83021020f7d6	af004276fd8f6c305727183c159b83021020f7d6
Author: Julian Smith <jules@op59.net>
Date:   Mon Nov 4 16:12:04 2019 +0000

    Bug 701829: ep_print_image(): protect against memcmp() beyond buffer.
    
    Fixes:
        ./sanbin/gs -dBATCH -dNOPAUSE -r12 -sOutputFile=tmp -sDEVICE=escp ../bug-701829.pdf

diff --git a/devices/gdevcdj.c b/devices/gdevcdj.c
index 8cf4e2e77..1ab1c202b 100644
--- a/devices/gdevcdj.c
+++ b/devices/gdevcdj.c
@@ -1969,7 +1969,7 @@ ep_print_image(gp_file *prn_stream, ep_globals *eg, char cmd, byte *data, int si
           /* p2 is the head of non zero image. */
           p2 = p3;
         redo:
-          for (p3 += row_bytes; memcmp(p3, zeros, row_bytes); p3 += row_bytes);
+          for (p3 += row_bytes; p3 < outp && memcmp(p3, zeros, row_bytes); p3 += row_bytes);
           if (p3 < outp && memcmp(p3+row_bytes, zeros, row_bytes)) goto redo;
         } else p1 = p2 = outp;
 
